packages/composer/amazeelabs/silverback_cloudinary/graphql/silverback_cloudinary.base.graphqls

```diff
@@ -0,0 +1,47 @@
+input ResponsiveImageVariant {
+  # Media query that determines when this variant should be used.
+  media: String!
+  # Default image dimensions. If height is omitted, the image is scaled, but
+  # not cropped.
+  width: Int!
+  height: Int
+  # A list of screen width / image width tuples that translate to a
+  # max-width and an image width that should be up until that screen width.
+  #
+  # Example:
+  # sizes: [
+  #  [400, 390] -> up until 400px screen width, use the 390px image
+  #  [800, 780] -> up until 800px screen width, use the 780px image
+  # ]
+  sizes: [[Int!]!]
+  # Additional cloudinary transformations.
+  # https://cloudinary.com/blog/on_the_fly_image_manipulations_secured_with_signed_urls
+  transform: String
+}
+
+input ResponsiveImageConfig {
+  # See above.
+  width: Int!
+  height: Int
+  sizes: [[Int!]!]
+  transform: String
+  # An array of responsive image variants.
+  variants: [ResponsiveImageVariant!]
+}
```

Comment:
If we drop the `variants` property, we could get away without an input type, since all others are primitives that could be passed as field arguments directly. That in turn would - in combination with the comment below - make the schema extension obsolete.

Reasoning behind it - variants have two purposes:

1. Format negotiation - selecting if webp & friends can be used based on the device.
2. Art direction - displaying entirely different images, based on the viewport.

Reason 1. is already solved by cloudinary's automatic format negotiation, and 2. can also be solved on a CSS / component level, where all other responsive behaviour is handled. One could argue that thats the better place to do that anyway.

Reply:
variants have now been dropped, and we don't have anymore new graphql types.